File: test-suite/src/test/java/io/micronaut/microstream/docs/PersistentCacheTest.java
package io.micronaut.microstream.docs;

import io.micronaut.context.ApplicationContext;
import io.micronaut.core.util.CollectionUtils;
import io.micronaut.runtime.server.EmbeddedServer;
import org.junit.jupiter.api.Test;

import java.util.Map;
import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;

class PersistentCacheTest {

    @Test
    void cachePersistsOverRestarts() {
        Map<String, Object> config = CollectionUtils.mapOf(
            "storageDirectory", "build/microstream-cache-" + UUID.randomUUID()
        );
        // When we create the app, and use a cached method
        try (EmbeddedServer server = ApplicationContext.run(EmbeddedServer.class, config, "cachepersist")) {
            CounterService counter = server.getApplicationContext().getBean(CounterService.class);
            counter.setCount("Tim", 1337L);
            Long count = counter.currentCount("Tim");
            assertEquals(1337L, count);
            counter.setCount("Tim", 666L);
        }

        // Then restarting the app with the same storage location, the value is still cached
        try (EmbeddedServer server = ApplicationContext.run(EmbeddedServer.class, config, "cachepersist")) {
            CounterService counter = server.getApplicationContext().getBean(CounterService.class);
            Long count = counter.currentCount("Tim");
            assertEquals(666L, count);
        }
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/microstream/docs/PersistentCacheTest.kt
package io.micronaut.microstream.docs

import io.micronaut.context.ApplicationContext
import io.micronaut.runtime.server.EmbeddedServer
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.util.UUID

class PersistentCacheTest {

    @Test
    fun cachePersistsOverRestarts() {
        val config = mapOf("storageDirectory" to "build/microstream-cache-${UUID.randomUUID()}")
        ApplicationContext.run(EmbeddedServer::class.java, config, "cachepersist").use {
            val counter = it.applicationContext.getBean(CounterService::class.java)
            counter.setCount("Tim", 1337)
            val count = counter.currentCount("Tim")
            assertEquals(1337, count)
            counter.setCount("Tim", 666)
        }
        ApplicationContext.run(EmbeddedServer::class.java, config, "cachepersist").use {
            val counter = it.applicationContext.getBean(CounterService::class.java)
            val count = counter.currentCount("Tim")
            assertEquals(666, count)
        }
    }
}
